File: kuba-front/src/app/shared/models/incident-create-request.ts
import IncidentType from "../enums/incident-type.enum";
import SeverityType from "../enums/severity.enum";

export interface IncidentCreateRequest {
    title: string;
    description: string;
    severityType: SeverityType;
    incidentType: IncidentType;
    reportedDate: Date
    userId: string;
}

File: kuba-back/Kuba.Api/Kuba.Api/Dtos/Incident/IncidentCreateRequest.cs
using Kuba.Domain.Enums;
using System.ComponentModel.DataAnnotations;

namespace Kuba.Api.Dtos.Incident
{
    public class IncidentCreateRequest
    {
        [Required]
        public string Title { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public DateTime ReportedDate { get; set; }
        [Required]
        public SeverityType SeverityType { get; set; }
        [Required]
        public IncidentType IncidentType { get; set; }

        [Required]
        public Guid UserId { get; set; }
    }
}
